src/iterator.cpp

```diff
@@ -42,7 +42,7 @@ template <typename T> std::vector<std::string> iterator_base<T>::get_current_pat
         return path_;
     }
 
-    std::vector<std::string> keys = path_;
+    memory::vector<memory::string> keys = path_;
     keys.reserve(path_.size() + stack_.size());
```

Comment:
You're causing a reallocation of the array inside the vector. Better to reserve and then copy from the two sources.